### PR TITLE
[#249][Feature] Github Actions Instructions for iOS Templates

### PR DESCRIPTION
**Pull request by blyscuit**
https://github.com/nimblehq/ios-templates/issues/249

## What happened

Add to Wiki

- Service to host the Github Actions runners (MacStadium).
- The setup guidelines (For MacStadium, Github Actions pipelines).
- The project setup guidelines with self-hosted runners with Github Actions.

### CODE REVIEW

File: .github/wiki/Self-Hosted-Github-Actions.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+![New Self Hosted Runner Script](assets/images/self-hosted-github-actions/new-self-hosted-runner-script.png)
+
+The general step are:
+
+1. Download: Download the newest version of Github Runner and save in the runner folder.
+2. Configure: Inject the project’s token to the runner and run the executable.
+
+## Add More Runners
+
+To add more runners to the same or a new host, simplify follow the instruction from repository’s settings `Settings > Actions > Runners > New self-hosted runner`. Make sure the new runner is downloaded to a different folder than the one before.****
```
Comment (suho) [inline .github/wiki/Self-Hosted-Github-Actions.md:52]: I think we can remove the asterisk
```suggestion
To add more runners to the same or a new host, simplify follow the instruction from repository’s settings `Settings > Actions > Runners > New self-hosted runner`. Make sure the new runner is downloaded to a different folder than the one before.
```

File: .github/wiki/Self-Hosted-Github-Actions.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## Restarting Runner
+
+In the case that the machine is shut down or the terminal running the script was an exit, the runner script can be relaunched by `cd` to the script folder and run `./run.sh`.****
```
Comment (suho) [inline .github/wiki/Self-Hosted-Github-Actions.md:56]: Same here, we can remove the asterisk
```suggestion
In the case that the machine is shut down or the terminal running the script was an exit, the runner script can be relaunched by `cd` to the script folder and run `./run.sh`.
```

File: .github/wiki/Self-Hosted-Github-Actions.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+### Remote/Rental Hosted Runner
+
+These are remote machines available to use as infrastructure-as-a-service or platform-as-a-service. They are more suitable to host the Github Actions runner as they are guaranteed to be online (as per the service’s terms). Covered in this guide is as [MacStadium](http://macstadium.com/). Other services include:
+
+- [https://aws.amazon.com/ec2/instance-types/mac/](https://aws.amazon.com/ec2/instance-types/mac/)
+- [http://www.macincloud.com/](http://www.macincloud.com/)
+
+## MacStadium Set Up
+
+A freshly rented MacStadium does not include all the tools needed to start a GitHub Actions for iOS development. The following steps are required to install all the dependencies.
```
Comment (vnntsu) [inline .github/wiki/Self-Hosted-Github-Actions.md:26]:
```suggestion
A freshly rented MacStadium does not include all the tools needed to start a GitHub Actions for iOS development. 

To install all the dependencies, follow these required steps:
```

Comment (blyscuit) [reply .github/wiki/Self-Hosted-Github-Actions.md:26]: nice.

File: .github/wiki/Self-Hosted-Github-Actions.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+1. Download: Download the newest version of Github Runner and save in the runner folder.
+2. Configure: Inject the project’s token to the runner and run the executable.
+
+## Add More Runners
+
+To add more runners to the same or a new host, simplify follow the instruction from repository’s settings `Settings > Actions > Runners > New self-hosted runner`. Make sure the new runner is downloaded to a different folder than the one before.
+
+## Restarting Runner
+
+In the case that the machine is shut down or the terminal running the script was an exit, the runner script can be relaunched by `cd` to the script folder and run `./run.sh`.
```
Comment (markgravity) [inline .github/wiki/Self-Hosted-Github-Actions.md:58]: Is it `the terminal running the script was exited`? 🤔

**Review by vnntsu — APPROVED** [review]
LGTM! 🚀

File: .github/wiki/Self-Hosted-Github-Actions.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+![New Self Hosted Runner Script](assets/images/self-hosted-github-actions/new-self-hosted-runner-script.png)
+
+The general steps are:
+
+1. Download: Download the newest version of Github Runner and save in the runner folder.
+2. Configure: Inject the project’s token to the runner and run the executable.
+
+## Add More Runners
+
+To add more runners to the same or a new host, simplify follow the instruction from repository’s settings `Settings > Actions > Runners > New self-hosted runner`. Make sure the new runner is downloaded to a different folder than the one before.
```
Comment (minhnimble) [inline .github/wiki/Self-Hosted-Github-Actions.md:54]: small improvements 🙏
```suggestion
To add more runners to the same or a new host, simply follow the instruction from repository’s settings `Settings > Actions > Runners > New self-hosted runner`. Make sure the new runner is downloaded onto a separated folder.
```

File: .github/wiki/Self-Hosted-Github-Actions.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+GitHub Actions allow the developer to set up a self-hosted runner in place of a Github-provided runner. A self-hosted runner can be a local device or a cloud device.
+
+### Local Hosted Runner
+
+Existing macOS devices can be used to run Github Actions Workflow by installing the runner and having the machine listen to the command from Github Actions.
+
+### Remote/Rental Hosted Runner
+
+These are remote machines available to use as infrastructure-as-a-service or platform-as-a-service. They are more suitable to host the Github Actions runner as they are guaranteed to be online (as per the service’s terms). Covered in this guide is as [MacStadium](http://macstadium.com/). Other services include:
```
Comment (minhnimble) [inline .github/wiki/Self-Hosted-Github-Actions.md:19]: Small improvements 
```suggestion
These are remotely available machines for using as infrastructure-as-a-service or platform-as-a-service. They are more suitable to host the Github Actions runner as they are guaranteed to be online (as per the service’s terms). Covered in this guide is as [MacStadium](http://macstadium.com/). Other services include:
```

Comment (blyscuit) [reply .github/wiki/Self-Hosted-Github-Actions.md:19]: How about
```suggestion
These are remote computers, available to use as infrastructure-as-a-service or platform-as-a-service. They are more suitable to host the Github Actions runner as they are guaranteed to be online (as per the service’s terms). Covered in this guide is as [MacStadium](http://macstadium.com/). Other services include:
```

**Review by markgravity — APPROVED** [review]
LGTM 🎉

**Review by suho — APPROVED** [review]

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

**Pull request merged**